common/connection.go

```diff
@@ -348,7 +348,16 @@ func (c *Connection) recvLoop() {
 			evt := ev.(*target.EventDetachedFromTarget)
 			sid := evt.SessionID
 			tid := c.findTargetIDForLog(sid)
-			c.closeSession(sid, tid)
+			ok := c.closeSession(sid, tid)
+			if !ok {
+				c.logger.Debugf(
+					"Connection:recvLoop:EventDetachedFromTarget",
+					"sid:%v tid:%v wsURL:%q, session not found",
+					sid, tid, c.wsURL,
+				)
+
+				continue
```

Comment:
Yes, I did some local checks before adding this `continue` here and have checked all those conditions. 

> If the emit of an event doesn't occur then that could cause a goroutine to block indefinitely such as

That shouldn't happen because we never let the browser attach unrelated pages. This will come from the browser for a page that didn't get attached. This is similar to the answer below.

> how does this work with the actual fix to prevent iterations from tracking each others browserContexts?

Good question :) The answer is the browser will keep sending these events if we don't attach the pages from other browser contexts. Here, we're ignoring its unnecessary demands to improve performance.